homeassistant/components/lifx/sensor.py

```diff
@@ -0,0 +1,208 @@
+"""Sensor entities for LIFX integration."""
+from __future__ import annotations
+
+from abc import abstractmethod
+from datetime import timedelta
+import logging
+from typing import cast
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TIME_MINUTES
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_RSSI,
+    DOMAIN,
+    HEV_CYCLE_DURATION,
+    HEV_CYCLE_LAST_POWER,
+    HEV_CYCLE_LAST_RESULT,
+    HEV_CYCLE_REMAINING,
+)
+from .coordinator import LIFXUpdateCoordinator
+from .util import lifx_features
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+RSSI_SENSOR = SensorEntityDescription(
+    key=ATTR_RSSI,
+    name="RSSI",
+    entity_category=EntityCategory.DIAGNOSTIC,
+    state_class=SensorStateClass.MEASUREMENT,
+    device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+    has_entity_name=True,
+    icon="mdi:wifi",
+)
+
+HEV_SENSORS = [
+    SensorEntityDescription(
+        key=HEV_CYCLE_DURATION,
+        name="Clean Cycle Duration",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=None,
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=TIME_MINUTES,
+        icon="mdi:timer",
+    ),
+    SensorEntityDescription(
+        key=HEV_CYCLE_REMAINING,
+        name="Clean Cycle Remaining",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=TIME_MINUTES,
+        icon="mdi:timer",
```

Comment:
```suggestion
```

We normally don't set an `icon` if the `device_class` is set. https://developers.home-assistant.io/docs/core/entity#advanced-properties